internal/reconciler/tracepipeline/reconciler.go

```diff
@@ -191,5 +191,14 @@ func (r *Reconciler) doReconcile(ctx context.Context, pipeline *telemetryv1alpha
 		return fmt.Errorf("failed to create otel collector metrics service: %w", err)
 	}
 
+	networkPolicyPorts := makeNetworkPolicyPorts()
+	networkPolicy := collectorresources.MakeNetworkPolicy(r.config, networkPolicyPorts)
+	if err = controllerutil.SetOwnerReference(pipeline, networkPolicy, r.Scheme()); err != nil {
```

Comment:
Add a watch for the resource during the controller setup in `tracepipeline_controller.go` to also have reconciliations triggered by modifications on the owned object. Same for the MetricPipeline.